### PR TITLE
Able to work with styled components. Fix infinite recursion

### PR DESCRIPTION
**Pull request by 13rentgen**
Able to work with styled components.
Fix infinite recursion.
Fix type error when a component not found.

#33 
#36

### CODE REVIEW

File: tests/__mocks__/vdom.js [hunk 1]
```diff
@@ -101,6 +101,76 @@ export const tree = {
...
+            children: [
+                {
+                    name: {
+                        componentStyle: {
+                            rules: [],
+                            isStatic: false,
+                            componentId: 'styled__Button-sc-1fuu6r1-1',
+                        },
+                        displayName: 'styled__Button',
+                        styledComponentId: 'styled__Button-sc-1fuu6r1-1',
```
Comment (baruchvlz) [inline tests/__mocks__/vdom.js]: I see that you add the `displayName` property in the util function, but where does the `styledComponentId` and `componentStyle` properties come from?

**Comment by baruchvlz** [comment]
One other thing, could you add a test for a case like this:

```js
resq$('MyComponent styled__Div')
resq$('FooComponent styled__Div OtherComponent')
```

File: tests/__mocks__/vdom.js (hunk 1 above)
Comment (13rentgen) [reply tests/__mocks__/vdom.js]: This properties come from styled components but do not need for the test actually

**Comment by 13rentgen** [comment]
> One other thing, could you add a test for a case like this:
> 
> ```js
> resq$('MyComponent styled__Div')
> resq$('FooComponent styled__Div OtherComponent')
> ```

Good thing, I found a bug :) 
Fix it and add tests.

By the way, this is playgrounds with Material UI lib: 
one with current branch version - https://stackblitz.com/edit/resq-alh9lx?file=index.js
And another one with latest release - https://stackblitz.com/edit/resq-7zy945?file=index.js

**Comment by baruchvlz** [comment]
I'll have to verify if and how this affects the integration with WebdriverIO. Once this is verified, I'll merge the PR and release.

File: src/utils.js [hunk 2]
```diff
@@ -191,6 +191,20 @@ export function buildNodeTree(element) {
...
+function findNode(children) {
+    while (children.length) {
+        const child = children.shift()
+
+        if (child.node) {
+            return child.node
+        }
+
+        if (child.children && Array.isArray(child.children)) {
+            children.push(...child.children)
```
Comment (simo-eskalera) [inline src/utils.js]: this findNode() method is destructive. It modifies the `children` parameter that's passed on to it. You could instead, just recursively call findNode() on the children node until a node is found

Comment (13rentgen) [reply src/utils.js]: findNode method using a copy of children, so original child not affected by the method

https://github.com/baruchvlz/resq/pull/37/files#diff-2b4ca49d4bb0a774c4d4c1672d7aa781R227

**Comment by 13rentgen** [comment]
> Also, could you update the documentation to mention the changes? How to select styled components and MatUI components.

There is no something special to find styled or MaterialUI components. I just using [React Devtools](https://github.com/facebook/react-devtools) which mentioned on Webdriver IO blog page here: https://webdriver.io/blog/2019/04/03/react-selectors.html

Here is React Devtools screenshot for this test case: 
https://stackblitz.com/edit/resq-alh9lx?file=index.js

Screenshot 2019-08-16 at 19 35 17

**Comment by baruchvlz** [comment]
In your example you're using `Styles(MuiBox)` as the selector for the MaterialUI Box component, if you do `resq$('MuiBox')` it doesn't work. This is has to be documented so the user understands how to properly select these components.

**Comment by baruchvlz** [comment]
I will do a release for this on Monday.

**Comment by simo-eskalera** [comment]
Thanks for enhancing the lookup to support HOC components.

**Pull request merged**